src/lib/components/button/QBtn.svelte

```diff
@@ -39,8 +43,22 @@
 
   const src = $derived(extractImgSrc(icon));
 
+  const finalVariant = $derived<typeof variant>(
+    variant !== "elevated"
+      ? variant
+      : filled
+        ? "filled"
+        : tonal
+          ? "tonal"
+          : outlined
+            ? "outlined"
+            : flat
+              ? "flat"
+              : "elevated"
+  );
+
   const color = $derived(
-    `var(--${design === "filled" ? "on-primary" : design === "tonal" ? "on-secondary-container" : "primary"})`
+    `var(--${finalVariant === "filled" ? "on-primary" : finalVariant === "tonal" ? "on-secondary-container" : "primary"})`
```

Comment:
Please always use braces for nested ternaries (or even better, avoid nested ternaries completely: https://eslint.org/docs/latest/rules/no-nested-ternary ). Maybe there's also an ESLint or Prettier rule for that.